app/api/thread.py

```
@@ -40,26 +47,32 @@ def thread_to_response(thread: Thread) -> ThreadResponse:
 
 
 @router.get("/stale", response_model=list[ThreadResponse])
-def list_stale_threads(
+async def list_stale_threads(
     current_user: Annotated[User, Depends(get_current_user)],
-    db: Session = Depends(get_db),
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)